### ISSUE TITLE
Parallel runs

### ISSUE THREAD

**Issue by BenjaminPeter** [gbradburd/SpaceMix#8]
When running spacemix on the same data (with different replicates / different models) at the same time, I found that the runs somehow interact with each other, with all runs result in nearly the same output figures, some of which are clearly inconsistent with the parameters of the particular run.

Is there a list of temporary files/folders that are written somewhere? Or some guidelines how spacemix should be configured to avoid these interactions?

**Comment by petrelharp** [comment]
Which functions are you calling to run spacemix?

**Comment by BenjaminPeter** [comment]
`run_spacemix_analysis`, a typical call is

```
run.spacemix.analysis(
                        fast.model.option = model,
                        long.model.option = model,
                        data.type = "counts",
                        counts = cts,
                        sample.sizes = ss,
                        spatial.prior.X.coordinates = pop_geo$longitude,
                        spatial.prior.Y.coordinates = pop_geo$latitude,
                        round.earth = TRUE,
                        k = nrow(cts),
                        loci = ncol(cts),

                        ###production option set
                        ngen = 1e7,
                        samplefreq = 5e4,
                        fast.MCMC.ngen = 5e5,
                        savefreq = 1e6,
                        n.fast.reps = 10,

                        ###debug option set
                        #ngen = 1e5,
                        #samplefreq = 1e3,
                        #fast.MCMC.ngen = 1e3,
                        #savefreq = 1e5,
                        #n.fast.reps = 3,

                        mixing.diagn.freq = 50,
                        printfreq = 1e5,
                        directory=output_name,
                        prefix = "_")
```

**Comment by gbradburd** [comment]
Is it possible that you're not changing the output_name you're specifying for the directory?  If so, SpaceMix will write over the existing files.

Also, in what way is the output inconsistent with the parameters of a particular run?

**Comment by petrelharp** [comment]
A good code snippet, that I'm tempted to make default for `directory`, is
```
directory=sprintf("run_%0.0f",1e6*runif(1))
```

Also, @gbradburd, looking at `run.spacemix.analysis`, if `prefix` is not specified, won't it throw an error [here](https://github.com/gbradburd/SpaceMix/blob/master/R/space.mix.MCMC.R#L1502)? Should it have default `prefix=''`?

**Comment by gbradburd** [comment]
Ok yeah @petrelharp I like both ideas and I've implemented them and pushed the updated package.  Prefix default is "MyRun" because otherwise all the output files start with "_", and that's silly.

**Comment by BenjaminPeter** [comment]
@petrelharp OK, I think I was just being dumb, or set up my scripts wrong somehow; I apologize. Essentially I got the exact same plot out of all four models yesterday, but deleted the files. I tried to reproduce this just now (on popres) and results varied (slightly) between models, so I suppose everything is running ok. I will reopen the issue if I re-encounter an example showing the issue I thought to have.
